File: SonicXBackendC/SonicXBackendC/Server/PlayList/PlaylistShowContent.cs
using Newtonsoft.Json.Linq;
using Newtonsoft.Json;
using SonicXBackendC.MusicManagment;
using SonicXBackendC.PlaylistManagment;
using SonicXBackendC.UserManagement;
using System.Collections.Generic;
using System.IO;
using System.Net;


namespace SonicXBackendC.Server.PlayList
{
    public class PlaylistShowContent
    {
        public void Handle(HttpListenerContext context)
        {
            if (context.Request.HttpMethod.Equals("POST"))
            {
                var objectMapper = new JsonSerializer();
                var userService = UserService.GetInstance();

                using (var reader = new StreamReader(context.Request.InputStream))
                {
                    var requestBody = reader.ReadToEnd();
                    var jsonRequest = JObject.Parse(requestBody);
                    var playlistName = jsonRequest["playlistName"].ToString();

                    Playlist playlist = new Playlist(playlistName);
                    List<MusicFile> musicFiles = playlist.GetAllMusicFiles(userService);

                    var responseJson = new JObject();
                    var dataArray = new JArray();

                    foreach (MusicFile musicFile in musicFiles)
                    {
                        var musicNode = new JObject();
                        musicNode["title"] = musicFile.Title;
                        musicNode["playlistName"] = playlistName;
                        musicNode["artist"] = musicFile.Artist;
                        musicNode["album"] = musicFile.Album;
                        musicNode["path"] = musicFile.Path;

                        dataArray.Add(musicNode);
                    }

                    responseJson["musicFiles"] = dataArray;

                    var response = System.Text.Encoding.UTF8.GetBytes(responseJson.ToString());
                    context.Response.ContentType = "application/json";
                    context.Response.StatusCode = (int)HttpStatusCode.OK;
                    context.Response.OutputStream.Write(response, 0, response.Length);
                }
            }
            else
            {
                context.Response.StatusCode = (int)HttpStatusCode.MethodNotAllowed;
            }
        }

    }
}

File: SonicXBackned/src/main/java/org/example/Server/Playlist/PlaylistShowContent.java
package org.example.Server.Playlist;


import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.example.MusicManagment.MusicFile;
import org.example.PlaylistManagment.Playlist;
import org.example.PlaylistManagment.PlaylistDAO;
import org.example.UserManagment.UserService;

import java.io.IOException;
import java.io.OutputStream;
import java.util.List;

public class PlaylistShowContent implements HttpHandler {
    public void handle(HttpExchange exchange) throws IOException {
        if ("POST".equals(exchange.getRequestMethod())) {
            ObjectMapper objectMapper = new ObjectMapper();
            UserService userService = UserService.getInstance();

            String requestBody = new String(exchange.getRequestBody().readAllBytes());
            ObjectNode jsonRequest = (ObjectNode) objectMapper.readTree(requestBody);
            String playlistName = jsonRequest.get("playlistName").asText();





            Playlist playlist = new Playlist(playlistName);
            List<MusicFile> musicFiles = playlist.getAllMusicFiles(userService);


            ObjectNode responseJson = objectMapper.createObjectNode();
            ArrayNode dataArray = objectMapper.createArrayNode();

            for (MusicFile musicFile : musicFiles) {
                ObjectNode musicNode = objectMapper.createObjectNode();
                musicNode.put("title", musicFile.getTitle());
                musicNode.put("playlistName", playlistName);
                musicNode.put("artist", musicFile.getArtist());
                musicNode.put("album", musicFile.getAlbum());
                musicNode.put("path", musicFile.getPath());


                dataArray.add(musicNode);
            }

            responseJson.put("musicFiles", dataArray);


            byte[] response = responseJson.toString().getBytes();
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, response.length);
            OutputStream os = exchange.getResponseBody();
            os.write(response);
            os.close();
        } else {
            exchange.sendResponseHeaders(405, -1); // Method Not Allowed
        }
    }

}

